### PR TITLE
{chem}[foss/2021a] LAMMPS v23Jun2022 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by ocaisa**
(created using `eb --new-pr`)
Requires https://github.com/easybuilders/easybuild-easyblocks/pull/2213

### CODE REVIEW

**Comment by ocaisa** [comment]
Test report by @ocaisa
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/99d4c0a83e3ee2b3e10830f14ba87a4e for a full test report.

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2021a-CUDA-kokkos.eb [hunk 1]
```diff
@@ -0,0 +1,166 @@
+name = 'LAMMPS'
+version = '23Jun2022'
+versionsuffix = '-CUDA'
+versionsuffix += '-kokkos'
```
Comment (jfgrimm) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2021a-CUDA-kokkos.eb:4]: as far as I'm aware, all easyconfigs include the CUDA version in the suffix
```suggestion
_cuda_suffix = '-CUDA-%(cudaver)s'
versionsuffix = _cuda_suffix + '-kokkos'
```

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2021a-CUDA-kokkos.eb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+builddependencies = [
+    ('CMake', '3.20.1'),
+    ('pkg-config', '0.29.2'),
+    ('archspec', '0.1.2'),
+]
+
+dependencies = [
+    ('CUDA', '11.3.1', '', SYSTEM),
+    ('NCCL', '2.10.3', '-CUDA-11.3.1'),
```
Comment (jfgrimm) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2021a-CUDA-kokkos.eb:36]:
```suggestion
    ('NCCL', '2.10.3', _cuda_suffix),
```

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**FAILED**
Build succeeded for 6 out of 9 (2 easyconfigs in total)
gpu02.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), 1 x NVIDIA Tesla V100-SXM2-32GB, 510.47.03, Python 3.6.8
See https://gist.github.com/d330c2ccd7eb83743db510184623bc0c for a full test report.

**Comment by ocaisa** [comment]
@jfgrimm The failure in your test was due to not being able to download the `kim-api` sources. Not sure why that happened, it's working fine for me today, glitch?

**Comment by ocaisa** [comment]
Test report by @ocaisa
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/503bbaf075945ae3c42ffedd68b2c950 for a full test report.

**Comment by ocaisa** [comment]
I think I will split out the CUDA build as I would like to fix the issue with CUDA-aware MPI and that doesn't need to hold this PR (or the easyblock PR) back

**Comment by bedroge** [comment]
> I copied the `kim-api` source tarball manually, now the installation fails at the sanity check:
> 
> ```
> OSError: libhdf5_hl.so.100: cannot open shared object file: No such file or directory
> ```
> 
> I don't get why it's expecting to find that file, as the libraries in the HDF5 installation all have `.200`. The OS itself has a much older version, and its libraries end with `.8`...

Ah, found it: `netCDF` is pulling in another HDF5 version... can we stick to `1.10.7` for `LAMMPS`, or does it need this newer version?

**Comment by ocaisa** [comment]
@bedroge Thanks for catching that, will fix now

**Comment by bedroge** [comment]
Test report by @bedroge
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
pg-interactive.hpc.rug.nl - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 3.6.8
See https://gist.github.com/a614d11ea59c9ed0f97f33f680c19544 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/02de446f21f7f9b2704e37c1dd7dbbca for a full test report.

**Comment by ocaisa** [comment]
@branfosj Your failure should be fixed with the latest commit

**Comment by ocaisa** [comment]
Test report by @ocaisa
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/83e1d70548354a4980f51d2de1e864a6 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/61f179370f42464376d271b4d4c5d75d for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/9f34827a71ee724a3d81a191baad0351 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2213
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/3b7be4027620ca7c765786f3bd3f4364 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by ocaisa** [comment]
@branfosj That failure is a download failure for Voro++

**Comment by ocaisa** [comment]
Ah, their cert is out of date

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2 EB_ARGS="--insecure-download"

**Comment by branfosj** [comment]
> @branfosj That failure is a download failure for Voro++

Should we add https://download.lammps.org/thirdparty/voro++-0.4.6.tar.gz as an alternative download location for Voro++?

**Comment by ocaisa** [comment]
@branfosj Done in #15898

**Review by branfosj — APPROVED** [review]
lgtm

**Comment by branfosj** [comment]
Going in, thanks @ocaisa!

**Pull request merged**